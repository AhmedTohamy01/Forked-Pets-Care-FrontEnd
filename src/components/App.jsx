import React from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import Home from "./Home";
import Dashboard from "./Dashboard";
import axios from "axios";
import "bootstrap/dist/css/bootstrap.css";
import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { LogIn, LogOut } from "../actions/index.js";
import Services from "../containers/Services";
import Landing from './Landing'
import '../styles/App.css';

const App = () => {
  const user = useSelector((state) => state.loggedInStatus);

  const dispatch = useDispatch();

  const checkLoginSatus = () => {
    axios
      .get("http://localhost:3001/logged_in", {
        withCredentials: true,
      })
      .then((response) => {
        console.log("login", response);

        if (
          response.data.logged_in &&
          user.loggedInStatus === "NOT_LOGGED_IN"
        ) {
          dispatch(LogIn(response.data.user));
        } else if (
          !response.data.logged_in &&
          user.loggedInStatus === "LOGGED_IN"
        ) {
          dispatch(LogOut());
        }
        // console.log('resp', response);
      })
      .catch((error) => {
        console.log(error);
      });
  };

  useEffect(() => {
    checkLoginSatus();
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <div className="App">
      <BrowserRouter>
        <Switch>
        <Route exact path={"/"} component={Landing} />
        <Route exact path={"/services"} component={Services} />
          <Route exact path={"/home"} component={Home} />
          <Route exact path={"/dashboard"} component={Dashboard} />
        </Switch>
      </BrowserRouter>
    </div>
  );
};

export default App;
