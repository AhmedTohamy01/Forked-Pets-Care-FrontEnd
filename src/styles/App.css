.LandingWrap {
  width: 100vw;
  height: 100vh;
  background: #101e42;
}

.LogoImage {
  background: url("../assets/logosmall.png");
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  width: 100vw;
  height: 40vw;
  margin: 2vh auto;
}

.DogVetImage {
  background: url("../assets/vet_dog.png");
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  width: 70vw;
  height: 70vw;
  margin: 0 auto;
}

.LandingTitle {
  font-family: 'Pacifico', cursive;

  /* font-family: 'Varela Round', sans-serif; */
  font-size: 3rem;
}

@media only screen
  and (min-device-width: 768px)
  and (max-device-width: 1024px)
  and (orientation: landscape) {
  .LandingWrap {
    height: 100vh;
  }

  .LogoImage {
    background: url("../assets/logosmall.png");
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    width: 40vw;
    height: 10vw;
    margin: 0 auto;
    margin-top: 5vh;
  }

  .DogVetImage {
    background: url("../assets/vet_dog.png");
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    width: 38vw;
    height: 36vw;
    margin: 0 auto;
  }
}

@media only screen
  and (min-device-width: 768px)
  and (max-device-width: 1024px)
  and (orientation: portrait) {
  .LandingWrap {
    height: 100vh;
  }

  .LogoImage {
    background: url("../assets/logosmall.png");
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    width: 70vw;
    height: 30vw;
    margin: 5vh auto;
  }

  .DogVetImage {
    background: url("../assets/vet_dog.png");
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    width: 38vw;
    height: 36vw;
    margin: 0 auto;
  }

  .LandingTitle {
    margin-top: 10vh;
  }
}

@media only screen
  and (min-device-width: 1024px)
  and (max-device-width: 1224px) {
  .LogoImage {
    background: url("../assets/logosmall.png");
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    width: 40vw;
    height: 10vw;
    margin: 0 auto;
    margin-top: 5vh;
  }

  .DogVetImage {
    background: url("../assets/vet_dog.png");
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    width: 25vw;
    height: 23vw;
    margin: 5vh auto;
  }
}

@media only screen
  and (min-width: 1224px) {
  .LandingWrap {
    height: 100vh;
  }

  .LogoImage {
    background: url("../assets/logosmall.png");
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    width: 40vw;
    height: 10vw;
    margin: 0 auto;
  }

  .DogVetImage {
    background: url("../assets/vet_dog.png");
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    width: 25vw;
    height: 23vw;
    margin: 2vh auto;
  }
}
