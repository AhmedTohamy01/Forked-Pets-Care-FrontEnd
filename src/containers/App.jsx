import React from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import Dashboard from "../components/Dashboard";
import axios from "axios";
import "bootstrap/dist/css/bootstrap.css";
import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { LogIn, LogOut } from "../actions/index.js";
import Services from "./Services";
import Landing from "../components/Landing";
import Login from "../components/auth/Login";
import Registration from "../components/auth/Registration";
import AdminRegistration from "../components/auth/AdminRegistration";
import "../styles/App.css";

const App = () => {
  const user = useSelector((state) => state.loggedInStatus);

  const dispatch = useDispatch();

  const checkLoginSatus = () => {
    axios
      .get("https://pets-care-api.herokuapp.com/logged_in", {
        withCredentials: true,
      })
      .then((response) => {
        if (
          response.data.logged_in &&
          user.loggedInStatus === "NOT_LOGGED_IN"
        ) {
          dispatch(LogIn(response.data.user));
        } else if (
          !response.data.logged_in &&
          user.loggedInStatus === "LOGGED_IN"
        ) {
          dispatch(LogOut());
        }
      })
      .catch((error) => {});
  };

  useEffect(() => {
    checkLoginSatus();
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <div className="App">
      <BrowserRouter>
        <Switch>
          <Route exact path={"/"} component={Landing} />
          <Route exact path={"/services"} component={Services} />
          <Route exact path={"/dashboard"} component={(user.loggedInStatus === "NOT_LOGGED_IN" ? Login : Dashboard)} />
          <Route exact path={"/login"} component={Login} />
          <Route exact path={"/signup"} component={Registration} />
          <Route exact path={"/adminsignup"} component={AdminRegistration} />
        </Switch>
      </BrowserRouter>
    </div>
  );
};

export default App;
