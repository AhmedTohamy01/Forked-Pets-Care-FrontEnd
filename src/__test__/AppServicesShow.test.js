import React from 'react';
import { render, fireEvent, cleanup } from './test-utils';
import '@testing-library/jest-dom';
import App from '../containers/App';

afterEach(cleanup);

describe('App', () => {
  test('should renderi main app', () => {
    expect(render(<App />)).toMatchSnapshot();
  });

  test('should render services buttons', () => {
    const { container } = render(<App />);
    expect(container.innerHTML).toMatch('Services');
  });

  test('should render Landing page properly', () => {
    const { container } = render(<App />);
    expect(container.innerHTML).toMatch('Taking care of your loved ones.');
  });

  test('should render service page', () => {
    const { container, getByTestId } = render(<App />);
    fireEvent.click((getByTestId('ServiceButton')));
    expect(container.innerHTML).toMatch('Check the differents services tha we offer.');
  });
});
