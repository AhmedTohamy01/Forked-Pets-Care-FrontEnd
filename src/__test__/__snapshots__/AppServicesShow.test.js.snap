// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renderi main app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="LandingWrap pt-2 "
        >
          <div>
            <div
              class="bm-overlay"
              style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
            />
            <div
              class="bm-menu-wrap"
              id=""
              style="position: fixed; z-index: 1100; width: 300px; height: 100%; transform: translate3d(-100%, 0, 0); transition: all 0.5s;"
            >
              <div
                class="bm-menu"
                style="height: 100%; box-sizing: border-box; overflow: auto;"
              >
                <nav
                  class="bm-item-list"
                  style="height: 100%;"
                >
                  <a
                    class="bm-item butt"
                    href="/services"
                    style="display: block;"
                    tabindex="-1"
                  >
                    Services
                  </a>
                  <a
                    class="bm-item my-2 butt"
                    href="/signup"
                    style="display: block;"
                    tabindex="-1"
                  >
                    Sign Up
                  </a>
                  <a
                    class="bm-item my-2 butt"
                    href="/login"
                    style="display: block;"
                    tabindex="-1"
                  >
                    Log In
                  </a>
                </nav>
              </div>
              <div>
                <div
                  class="bm-cross-button"
                  style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
                >
                  <span
                    style="position: absolute; top: 6px; right: 14px;"
                  >
                    <span
                      class="bm-cross"
                      style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
                    />
                    <span
                      class="bm-cross"
                      style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
                    />
                  </span>
                  <button
                    style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
                    tabindex="-1"
                  >
                    Close Menu
                  </button>
                </div>
              </div>
            </div>
            <div>
              <div
                class="bm-burger-button"
                style="z-index: 1000;"
              >
                <span>
                  <span
                    class="bm-burger-bars"
                    style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
                  />
                  <span
                    class="bm-burger-bars"
                    style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
                  />
                  <span
                    class="bm-burger-bars"
                    style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
                  />
                </span>
                <button
                  style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
                >
                  Open Menu
                </button>
              </div>
            </div>
          </div>
          <div
            class="LogoImage"
          />
          <div
            class="DogVetImage"
          />
          <h1
            class=" mb-2 text-white LandingTitle text-center"
          >
            Taking care of your loved ones.
          </h1>
          <div
            class="mt-5 mb-1 mt-md-5 mt-lg-4 pb-3 mx-auto text-center"
          >
            <a
              href="/services"
            >
              <span
                class="btn btn-light rounded-pill btn-lg"
                data-testid="ServiceButton"
                type="button"
              >
                Services
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="LandingWrap pt-2 "
      >
        <div>
          <div
            class="bm-overlay"
            style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
          />
          <div
            class="bm-menu-wrap"
            id=""
            style="position: fixed; z-index: 1100; width: 300px; height: 100%; transform: translate3d(-100%, 0, 0); transition: all 0.5s;"
          >
            <div
              class="bm-menu"
              style="height: 100%; box-sizing: border-box; overflow: auto;"
            >
              <nav
                class="bm-item-list"
                style="height: 100%;"
              >
                <a
                  class="bm-item butt"
                  href="/services"
                  style="display: block;"
                  tabindex="-1"
                >
                  Services
                </a>
                <a
                  class="bm-item my-2 butt"
                  href="/signup"
                  style="display: block;"
                  tabindex="-1"
                >
                  Sign Up
                </a>
                <a
                  class="bm-item my-2 butt"
                  href="/login"
                  style="display: block;"
                  tabindex="-1"
                >
                  Log In
                </a>
              </nav>
            </div>
            <div>
              <div
                class="bm-cross-button"
                style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
              >
                <span
                  style="position: absolute; top: 6px; right: 14px;"
                >
                  <span
                    class="bm-cross"
                    style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
                  />
                  <span
                    class="bm-cross"
                    style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
                  />
                </span>
                <button
                  style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
                  tabindex="-1"
                >
                  Close Menu
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              class="bm-burger-button"
              style="z-index: 1000;"
            >
              <span>
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
                />
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
                />
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
                />
              </span>
              <button
                style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0; background: transparent; cursor: pointer;"
              >
                Open Menu
              </button>
            </div>
          </div>
        </div>
        <div
          class="LogoImage"
        />
        <div
          class="DogVetImage"
        />
        <h1
          class=" mb-2 text-white LandingTitle text-center"
        >
          Taking care of your loved ones.
        </h1>
        <div
          class="mt-5 mb-1 mt-md-5 mt-lg-4 pb-3 mx-auto text-center"
        >
          <a
            href="/services"
          >
            <span
              class="btn btn-light rounded-pill btn-lg"
              data-testid="ServiceButton"
              type="button"
            >
              Services
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
